tracee-rules/signatures/rego/regosig/traceerego.go

```diff
@@ -62,13 +66,22 @@ func NewRegoSignature(regoCodes ...string) (types.Signature, error) {
 		return nil, err
 	}
 
-	res.matchPQ, err = rego.New(
-		rego.Compiler(res.compiledRego),
-		rego.Query(fmt.Sprintf(queryMatch, pkgName)),
-	).PrepareForEval(context.TODO())
-	if err != nil {
-		return nil, err
+	var rerr error
+	if aioEnabled {
+		res.matchPQ, rerr = rego.New(
+			rego.Compiler(res.compiledRego),
+			rego.Query(queryMatchAll),
+		).PrepareForEval(context.TODO())
```

Comment:
Note: I believe we need to rethink this as currently we create a Rego engine per signature and in the case of an AIO we need 1 engine with all rules with a selector (query) on the AIO rule. Thoughts and comments appreciated.

Reply:
I fixed this here: https://github.com/aquasecurity/tracee/pull/993/commits/b81da5a8ac88666105a8eae88fcb7d102c49123b